mllib/src/main/scala/org/apache/spark/mllib/optimization/GradientDescent.scala

```diff
@@ -178,7 +178,7 @@ object GradientDescent extends Logging {
       regParam: Double,
       miniBatchFraction: Double,
       initialWeights: Vector,
-      convergenceTol: Double): (Vector, Array[Double]) = {
+      convergenceTol: Double): (Vector, Array[Double], Integer) = {
```

Comment:
yes, I guess in this case # of iter = history length. I can remove the last argument